vectors/no_resizable_vec.rs

```diff
@@ -1,108 +1,93 @@
 extern crate alloc;
-use alloc::alloc::{Layout, alloc, dealloc, handle_alloc_error};
+use core::ptr::NonNull;
 use core::mem;
-use core::ops::{Deref, DerefMut, Index, IndexMut, Range};
-use core::ptr::{self, NonNull};
-use std::cmp::Ordering;
-use cvt::CVT_assume;
+use std::{ptr, ops::{Deref, DerefMut, Index, IndexMut}};
+use alloc::alloc::{Layout, alloc, dealloc};
 use nondet::Nondet;
 use std::io::Read;
 use anchor_lang::prelude::borsh::maybestd::io::Write;
 use borsh::{BorshDeserialize, BorshSerialize};
 
-////////////////////////////////////////////////////////////////////////
-// Adapted from SeaHorn.
-//
-// This is a more precise implementation of the Vector class than
-// NoDataVec but the vector is not resizable so its capacity must be
-// fixed a priori.
-////////////////////////////////////////////////////////////////////////
-
-trait IsZST { const IS_ZST: bool; }
-impl<T> IsZST for T { const IS_ZST: bool = mem::size_of::<T>() == 0; }
-
-macro_rules! assert {
-        ($cond:expr) => {{ /*cvt::CVT_assert($cond)*/}};
-}
-
+/////////////////////////
+/// Raw Vec
+/////////////////////////
 struct RawVec<T> {
     ptr: NonNull<T>,
     cap: usize,
 }
 
-unsafe impl<T: Send> Send for RawVec<T> {}
-unsafe impl<T: Sync> Sync for RawVec<T> {}
-
 impl<T> RawVec<T> {
-    // We assume that there is no errors during allocation.
-    fn new(capacity: usize) -> Self {
-        if T::IS_ZST || capacity == 0 {
-            return RawVec {
-                ptr: NonNull::dangling(),
-                cap: capacity,
-            };
+    fn new_zero_sized() -> Self {
+        Self {
+            ptr: NonNull::dangling(),
+            cap: 0,
         }
+    }
+
+    fn new(capacity: usize) -> Self {
 
-        let layout: Layout = Layout::array::<T>(capacity).unwrap();
-	    //CVT_assume(layout.size() <= isize::MAX as usize);
+        // ZSTs have no memory allocation
+        if mem::size_of::<T>() == 0 || capacity == 0 {
+            return RawVec::new_zero_sized();
+        }
 
-        let pointer: *mut u8 = unsafe { alloc(layout) };
-        let pointer: NonNull<T> = match NonNull::new(pointer as *mut T) {
-            Some(p) => p,
-            None => {
-                CVT_assume(false);
-                handle_alloc_error(layout)
-            },
+        let layout: Layout = Layout::array::<T>(capacity).unwrap_or_else(|_| panic!("capacity overflow"));
```

Comment:
the non-data vet should have same check for potential capacity overflow

Reply:
i do have an assert in `grow` :
```
        assert!(new_cap < usize::MAX);
```